### PR TITLE
Add Triggerer to Helm Chart

### PR DESCRIPTION
**Pull request by kaxil**
Adds triggerer component added in https://github.com/apache/airflow/pull/15389 ([AIP-40](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=177050929)) to the Helm Chart

ToDo:

- [x] Add tests
- [x] Add docs

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: chart/templates/triggerer/triggerer-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+              subPath: airflow_local_settings.py
+              readOnly: true
+            {{- end }}
+          envFrom:
+          {{- include "custom_airflow_environment_from" . | default "\n  []" | indent 10 }}
+          env:
+          {{- include "custom_airflow_environment" . | indent 10 }}
+          {{- include "standard_airflow_environment" . | indent 10 }}
+{{- if $persistence }}
+        - name: triggerer-log-groomer
```
Comment (jedcunningham) [inline chart/templates/triggerer/triggerer-deployment.yaml]: Right now the triggerer doesn't start the log server, should it?

I haven't really dug in here, but if we are storing logs locally, thus needing a groomer, don't we need to be able to get them?

File: chart/values.yaml [hunk 2]
```diff
@@ -715,6 +715,113 @@ webserver:
...
+    # The name of the ServiceAccount to use.
+    # If not set and create is true, a name is generated using the release name
+    name: ~
+
+    # Annotations to add to triggerer kubernetes service account.
+    annotations: {}
+
+  persistence:
+    # Enable persistent volumes
+    enabled: false
```
Comment (jedcunningham) [inline chart/values.yaml]: Should this default to True like workers?

File: chart/templates/triggerer/triggerer-deployment.yaml (hunk 1 above)
Comment (potiuk) [reply chart/templates/triggerer/triggerer-deployment.yaml]: Good point - should we surface the logs of triggerer in the UI somehow? are there any useful logs for DAG writers and ones that can be connected to particular task runs?

File: chart/templates/triggerer/triggerer-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,212 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (potiuk) [inline chart/templates/triggerer/triggerer-deployment.yaml:1]: Same as in docker compose, I believe this won't work for Airflow pre 2.2. But people will use Helm Chart for 2.1 and 2.0 versions of airflow for quite a while. Should we make some ifs here?

Comment (jedcunningham) [reply chart/templates/triggerer/triggerer-deployment.yaml:1]: The check is there, line 22 (currently).

Swapping $persistence and the version check would probably be a good idea though.

Comment (potiuk) [reply chart/templates/triggerer/triggerer-deployment.yaml:1]: Oh yeah. Shame on me :(

File: chart/values.yaml (hunk 2 above)
Comment (kaxil) [reply chart/values.yaml]: I made it a deployment -- there isn't any state on Triggers

File: chart/templates/triggerer/triggerer-deployment.yaml (hunk 1 above)
Comment (kaxil) [reply chart/templates/triggerer/triggerer-deployment.yaml]: Me and Andrew G had a talk this morning about it -- currently we don't store logs and there isn't much useful data in Triggerer logs but we will add more observability aroud Triggerer -- most likely somehow show it with Task Logs or a something similar, we haven't yet finalized.

File: chart/templates/triggerer/triggerer-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+                os.environ['AIRFLOW__LOGGING__LOGGING_LEVEL'] = 'ERROR'
+
+                from airflow.jobs.triggerer_job. import TriggererJob
+                from airflow.utils.db import create_session
+                from airflow.utils.net import get_hostname
+                import sys
+
+                with create_session() as session:
+                    job = session.query(TriggererJob).filter_by(hostname=get_hostname()).order_by(
+                        SchedulerJob.latest_heartbeat.desc()).limit(1).first()
```
Comment (kaxil) [inline chart/templates/triggerer/triggerer-deployment.yaml]:
```suggestion
                        TriggererJob.latest_heartbeat.desc()).limit(1).first()
```

File: chart/values.yaml [hunk 5]
```diff
@@ -719,6 +719,78 @@ webserver:
...
+triggerer:
+  # Number of airflow triggerers in the deployment
+  replicas: 1
+
+  # Command to use when running Airflow triggerers (templated).
+  command: ~
+  # Args to use when running Airflow triggerer (templated).
+  args: ["bash", "-c", "exec airflow triggerer"]
+
+  # Update Strategy when triggerers are deployed as a Deployment
```
Comment (jedcunningham) [inline chart/values.yaml]:
```suggestion
  # Update Strategy for triggerers
```

**Review by jedcunningham — APPROVED** [review]

**Pull request merged**